### DISCUSSION TITLE
What's the strategy for a `tags` field that's a `belongs_to` of another model?

### DISCUSSION THREAD

**Discussion by johlym** [avo-hq/avo#3396] [Help]
In my app, I have a `MenuItem` with a reusable `Style`, and said `Style` can have one or more `MenuItem`s. I'm at a loss as to how to implement this in the `MenuItem` resource (specifically as a `tags` field). Ideally, I'd like to feed the `suggestions` something like `Style.pluck(:name)`, limit it to 1 tag, and have, if new ones not on the suggestion list are added, a new `Style` is created.

It _seems_ like ActsAsTaggableOn might work here, but I haven't explored whether this concept of tagging stuff will expand to other attributes or models so I don't know if it's just overcomplication. 

Input appreciated!

**Comment by Paul-Bob** [comment]
Hi @johlym I'm not sure if I understand.

Why aren't you using `belongs_to` field and trying to use `tags`?

**Comment by johlym** [comment]
Thanks for replying @Paul-Bob ! Fair question, for sure.

Mostly the UI. `belongs_to` seems to limit me to a dropdown (unless I'm missing something)? There'll be cases where the `Style` doesn't exist, yet, and there's a higher cognitive load for folks creating new ones than simply typing in a new one into a `tags` field and having it magically exist on save.

**Comment by Paul-Bob** [comment]
> Mostly the UI. belongs_to seems to limit me to a dropdown (unless I'm missing something)?

On Avo pro and above, you can make the belongs to `searchable: true` ([docs here](https://docs.avohq.io/3.0/associations/belongs_to.html#searchable)) and it will behave like the course field here: https://main.avodemo.com/avo/resources/course_links/new

> and there's a higher cognitive load for folks creating new ones than simply typing in a new one into a tags field and having it magically exist on save.

If you only require a label to create a new style and `belongs_to` is not the right UI/UX for you, I think you can go in the `tags` direction.

---

I suggest starting by reading the whole [tags docs page](https://docs.avohq.io/3.0/fields/tags.html#tags-field), after that, you'll have an overview of all the possibilities and how to combine options to achieve your desired output.

You could use `select` mode ([docs here](https://docs.avohq.io/3.0/fields/tags.html#mode)) combined with `suggestions` if you have a small amount of styles or `fetch_values_from` on larger amounts of styles.

**Comment by johlym** [comment]
Thanks, @Paul-Bob! For now, I went in the direction of the tag with `ActsAsTaggableOn,` and a `tags` field, but it's not _ideal_ since it allows more than one tag, and there appears to be no way to enforce limits (ex: ensuring only one tag is assigned). That and it's a big hammer for a small nail, so to speak.

I'll take another look at `belongs_to`. Making it searchable gets me a lot closer to where I'd like to be, for sure. 

Ultimately, since I need to create relationships between the `MenuItem` and `Style` records (ex: `Style.where(name: 'stout').menu_items`), having static tags written to each `MenuItem` won't ultimately work. Bringing `ActsAsTaggableOn` into the picture closed that gap and helped build the association aspect of this, but like I said, big hammer.
